src/Symfony/Component/Yaml/Escaper.php

```diff
@@ -80,9 +88,14 @@ public static function requiresSingleQuoting($value)
             return true;
         }
 
-        // Determines if the PHP value contains any single characters that would
-        // cause it to require single quoting in YAML.
-        return preg_match('/[ \s \' " \: \{ \} \[ \] , & \* \# \?] | \A[ \- ? | < > = ! % @ ` ]/x', $value);
+        // First character is an indicator
+        // @see http://yaml.org/spec/1.2/spec.html#c-indicator
+        if ($value && 1 === strspn($value[0], '-?&*!|<>\'"=%@`')) {
```

Comment:
this is broken for the empty string (btw, the simpler fix for that is probably to add the empty string in the array of values above)